control_plane/attachment_service/src/reconciler.rs

```diff
@@ -114,11 +114,16 @@ impl Reconciler {
             .locations
             .insert(node.id, ObservedStateLocation { conf: None });
 
+        // 'lazy' mode means that a tenant's activiation will be subject to a concurrency limit
+        // in the pageserver: this is the safer way to attach a tenant, as it avoids thundering
+        // herd I/O when attaching many tenants at the same time.
+        let lazy = true;
```

Comment:
If I understand correctly, `Reconciler::live_migrate` will call with lazy=true as well and as we discussed just now (re: #7000) we should not do it.

I see two alternatives:
- complicate calling this fn
- always use SpawnMode::Eager on pageserver side

Reply:
Yep, it's easy enough to make use of lazy=true specific to the case where we want it, so I've just made this an argument to the function, and set it true for the "general case" part of Reconciler::reconcile (and false for live migrations)